daffodil-core/src/main/scala/org/apache/daffodil/grammar/ElementBaseGrammarMixin.scala

```diff
@@ -605,7 +605,12 @@ trait ElementBaseGrammarMixin
   }
 
   private lazy val textStandardNumber = prod("textStandardNumber", textNumberRep == TextNumberRep.Standard) {
-    ConvertTextCombinator(this, stringValue, textConverter)
+    val converter = textStandardBaseDefaulted match {
+      case 10 => textConverter
+      case 2 | 8 | 16 =>  textStandardNonBaseTenConverter
```

Comment:
I think we should support base 256 also, where the charcodes 0 to 255 are the digits.

Base e (2.71828...) has also been suggested somewhere.....

Just kidding.

Reply:
The DFDL spec is clear on what valid values are for textStandardBase, and they are only 2, 8, 10, 16.  I would want a compelling reason to deviate from this.

Base 256 sounds more like binary encoding; I'm not sure we gain much by allowing it as a textStandardBase as well. Eg. when the string "0" means 48, I do not think you are working with text.

Reply:
What's the use case for base-256? A base-256 encoded number would probably be full of non-display characters, so I'm not sure why that would ever be used in practice?

Implementation also becomes a bit more difficult, since right now we just rely on Java utils to converts to/from base-2,8,16 encoded numbers. Supporting anything means we have to implement our own logic since that's all java supports. Not hard, but more work and more code to maintain. Just want to make sure there's a good use case for it.

Reply:
@mbeckerle , thoughts on base 256? I think all other comments have been addressed and this commit is ready to merge otherwise.

Reply:
You missed the "Just Kidding" at the end of my comment. :-)

+1

Reply:
Ah, I thought you were just talking about base e. I'll merge this.